packages/vertexai/src/requests/response-helpers.ts

```diff
@@ -25,6 +25,26 @@ import {
 } from '../types';
 import { VertexAIError } from '../errors';
 
+/**
+ * Creates an EnhancedGenerateContentResponse object that has helper functions and
+ * other modifications that improve usability.
+ */
+export function createEnhancedContentResponse(
+  response: GenerateContentResponse
+): EnhancedGenerateContentResponse {
+  /**
+   * The Vertex AI backend omits default values.
+   * This causes the `index` property to be omitted from the first candidate in the
+   * response, since it has index 0, and 0 is a default value.
+   */
+  if (response.candidates && !response.candidates[0].index) {
```

Comment:
This will always kick in if response.candidates exists because if response.candidates[0].index === 0 then that's also falsy and fulfills the second condition. Which is fine I guess if somewhat redundant. Maybe just `if (response.candidates?.[0])` for the same effect. Or if you want to be super precise, make the second condition `response.candidates[0].index === undefined` or `!response.candidates[0].hasOwnProperty('index')`

Reply:
Thanks for catching this! Fixed